### ISSUE TITLE
Add a config option to skip a11y checks on element-based composition (`asChild` / `render`)

### ISSUE THREAD

**Issue by tobigumo** [jsx-eslint/eslint-plugin-jsx-a11y#1091]
### Summary
Element-based composition (Radix's `asChild`, Base UI / shadcn's `render`) produces false positives such as `anchor-has-content` and `control-has-associated-label` on valid code. Polymorphism via the `as` prop is already supported (#923), but element-based composition isn't covered.

### Minimal reproduction
```tsx
<Button render={<a href="/x" />}>Label</Button>
// false positive: anchor-has-content on <a/>
// (at runtime the wrapping component's children are forwarded → <a>Label</a>)

<Button asChild aria-label="Delete"><a href="/x" /></Button>
// false positive (at runtime the aria-label is forwarded to the child)
```
In both cases the wrapping component and the composed element merge into a single element at runtime, but the linter evaluates them as two separate nodes.

### Request
Add an optional setting to declare composition props (e.g. `asChild`, `render`). When a component uses one of them, it and its composed element are excluded from a11y evaluation. This would sit in the same family as `polymorphicPropName` / `components`. It is the user's responsibility to declare only components that actually forward props.

Fully resolving the merged element (instead of skipping it) could work in theory, but prop-merge semantics are library-specific (e.g. which value wins on conflict), so they can't be resolved generically. This issue only asks for the skip behavior.

### Optional follow-up (not required)
For rules that check for the presence of content or a label (`anchor-has-content`, `control-has-associated-label`, `alt-text`), these false positives could be *resolved* rather than skipped by checking the union of the wrapping component and the composed element. Forwarding is the contract of these patterns, and presence doesn't depend on which value wins a merge, so this stays valid without modeling any library-specific merge rules. Skipping alone already removes the false positives, so this is just a possible future enhancement.
